aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -2044,29 +2134,72 @@ public void onFailure(Exception exception) {
         };
     }
 
+    /**
+     * Used to reset password if user forgot the old password.
+     *
+     * @param username username of the user trying to reset password.
+     * @param clientMetadata meta data to be passed to the lambdas invoked by confirm sign up operation.
+     * @param callback callback will be invoked to notify the success or failure of the
+     *                 forgot password operation
+     */
+    @AnyThread
+    public void forgotPassword(final String username,
```

Comment:
Do we need clientMetadata for forgotPassword?

Reply:
ForgotPassword operation can potentially invoke pre-signup, user migration and custom message lambda triggers. Based on that custom logic in those lambdas, user might need to pass inputs to those lambda using client meta data. This was also the part of original request https://github.com/aws-amplify/aws-sdk-ios/issues/2299

Reply:
hey guys.. what's the status of this PR please?